src/wp-includes/class-wp-block.php

```diff
@@ -471,6 +471,12 @@ public function render( $options = array() ) {
 			}
 		}
 
+		if ( ! empty( $this->block_type->view_script_module_ids ) ) {
```

Comment:
I was wondering whether it would fit better after all filters get applied like `render_block`. This way, plugin authors still would have the opportunity to change the list of module IDs on the block.

Reply:
Extenders can add/remove/replace what is inside `$this->block_type->view_script_module_ids` with filters like `render_block`, so maybe it's worth accounting for it. Anyway, the same issue exists today with view scripts, so maybe we need to live with it for conistency.

Reply:
I'm happy to defer to your judgment. My instinct was to align with how scripts work but we do not strictly need to do if we can improve things.

Reply:
Maybe we can check if differs after `filters` and do a `_doing_it_wrong()` with the message of `Changing the module graph is error prone`

Reply:
I don't have a strong opinion about this but from my experience trying to change `supports.interactivity` dynamically, I don't think mutating the Block Type at render time should be encouraged. The way I see it now is that the Block Type is meant to represent the properties of all the blocks of that type, not just the one being rendered. Changing it on the fly to accommodate the block that is being rendered is confusing and leads to complexity. So, I think that the differences between the rendered blocks of the same type should be reflected by other means.

Reply:
Do you anticipate introducing any conditional loading for view script modules depending on the HTML rendered for the block? Let's say enqueueing only these registered modules for the block that has corresponding directives in the final HTML. That would be a more compelling reason to move the logic after block render filters.